File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/bridge/solution/RectangleBody.java
// RectangleBody.java

class RectangleBody extends RightAngledQuadrilateralBody
{
	public RectangleBody(double aHeight, double aWidth)
	{
		myHeight = aHeight;
		myWidth  = aWidth ;
	}

	public double getHeight()
	{
		return myHeight;
	}

	public double getWidth()
	{
		return myWidth;
	}

	public boolean isSquare()
	{
		return false;
	}

	private double myHeight;
	private double myWidth ;
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_cpp_labs_and_solutions/bridge/solution/rectangle_body.cpp
// rectangle_body.cpp

#include "rectangle_body.h"

rectangle_body::rectangle_body(double a_height, double a_width)
{
	my_height = a_height;
	my_width  = a_width ;
}

double rectangle_body::get_height() const
{
	return my_height;
}

double rectangle_body::get_width() const
{
	return my_width;
}

bool rectangle_body::is_square() const
{
	return false;
}
